### ISSUE TITLE
index.js requires lib/axios which does not respect pre built /dist/axios

### ISSUE THREAD

**Issue by weichx** [mzabriskie/axios#353]
requiring axios should only include a single module. currently require('axios') will pull in `lib/axios` which pulls in a bunch of other files. index.js should be changed to `require('./dist/axios')`

**Comment by nickuraltsev** [comment]
I'm not sure I understand what the problem is. Could you please elaborate?

**Comment by weichx** [comment]
right now if I `require('axios')` i get the following files included via webpack automatically:

  [94] ./~/axios/index.js 40 bytes {1} [built]
  [95] ./~/axios/lib/axios.js 3.45 kB {1} [built]
  [96] ./~/axios/lib/defaults.js 1.66 kB {1} [built]
  [97] ./~/axios/lib/utils.js 5.98 kB {1} [built]
  [98] ./~/axios/lib/core/dispatchRequest.js 946 bytes {1} [built]
  [99] ./~/axios/lib/adapters/xhr.js 4.08 kB {1} [optional] [built]
 [100] ./~/axios/lib/helpers/buildURL.js 1.49 kB {1} [built]
 [101] ./~/axios/lib/helpers/parseHeaders.js 784 bytes {1} [built]
 [102] ./~/axios/lib/helpers/transformData.js 550 bytes {1} [built]
 [103] ./~/axios/lib/helpers/isURLSameOrigin.js 2.25 kB {1} [built]
 [104] ./~/axios/lib/helpers/btoa.js 1.1 kB {1} [built]
 [105] ./~/axios/lib/helpers/cookies.js 1.36 kB {1} [built]
 [106] ./~/axios/lib/core/InterceptorManager.js 1.25 kB {1} [built]
 [107] ./~/axios/lib/helpers/isAbsoluteURL.js 563 bytes {1} [built]
 [108] ./~/axios/lib/helpers/combineURLs.js 348 bytes {1} [built]
 [109] ./~/axios/lib/helpers/bind.js 256 bytes {1} [built]
 [110] ./~/axios/lib/helpers/spread.js 564 bytes {1} [built]

when I change index.js to be `require('./dist/axios')`, I get the following (as expected)
 [94] ./~/axios/index.js 41 bytes {1} [built]
  [95] ./~/axios/dist/axios.js 30.6 kB {1} [built]

basically, the dist directory isnt used right now, it should be.

**Comment by mzabriskie** [comment]
If you are using node you will need `lib/axios.js` which is the default export from `import axios from 'axios'` as it will use the `http` adapter. For the browser `dist/axios.js` is ideal b/c it has `http` code path eliminated to save bytes. If you are bundling yourself with webpack/browserify you can still use the default `lib/axios.js`. Ultimately `dist/axios.js` is for installing from bower.
